### ISSUE TITLE
[FR] Support bare domains without any special configuration

### ISSUE THREAD

**Issue by jrpear** [docker-mailserver/docker-mailserver#3197]
# Feature Request

### Is your Feature Request related to a Problem?

Right now, you need to modify the postfix config in order to use a bare domain ([FAQ](https://jrpear.github.io/docker-mailserver/edge/faq/#can-i-use-a-nakedbare-domain-ie-no-hostname)). I'd like to remove that requirement.

The warning which a misconfigured deploy can produce,

```
do not list domain in BOTH mydestination and virtual_mailbox_domains
```

has been mentioned in 23 issues, 2 open. Scanning though a few of them, I think many were related to an email account being `@$myhostname` and thus delivered with the `local` delivery agent rather than the `virtual` delivery agent.

### Describe the Solution you'd like

I'd like to make the change described in that FAQ in the project's Postfix `main.cf`. So change

```
mydestination = $myhostname, localhost.$mydomain, localhost
```

to

```
mydestination = localhost.$mydomain, localhost
```

In fact, I don't know if those other two items are useful either... perhaps we should just unset `mydestination` altogether.

### Are you going to implement it?

Yes, because I know the probability of someone else doing it is low and I can learn from it.

### What are you going to contribute??

I'll contribute that change and modifications to relevant code. So I'd remove [this workaround for LDAP](https://github.com/docker-mailserver/docker-mailserver/blob/e890ba46a3e64ea0cc75b1cfd3d56e478655cd4b/target/scripts/startup/setup.d/ldap.sh#L93), the root->postmaster alias, and maybe more.

### Alternatives you've considered

I can't really think of another solution. For inspiration, I looked at [mailcow's main.cf](https://github.com/mailcow/mailcow-dockerized/blob/91906a711f7f8c4be93bdbde8f6e9d423485a54a/data/conf/postfix/main.cf#L174). They don't assign `$myhostname` to `mydestination`.

### Who will that Feature be useful to?

Anyone hosting on a bare domain. Possibly other people who have related issues that have gone stale.

### What have you done already?

I've found that this change (just removing `$myhostname`, or unsetting `mydestination` completely) breaks three unit tests:

- `[SMTP] (delivery) delivers mail to existing account`
- `[SMTP] (delivery) user1 should have received 8 mails`
- `[SMTP] (delivery) mail for root was delivered`

All three are broken because a locally sent message to `root` (with `sendmail`) isn't delivered correctly to `user1@localhost.localdomain`. I haven't investigated why yet.

Unsetting `mydestination` completely would prevent people from sending mail to system users of the container. But I can't imagine anyone does that now.

**Comment by georglauterbach** [comment]
@jrpear do you think you will be able to get this done in the near future? I'm asking because I#m trying to get to v12.1.0 fast, and if this issue takes a while to tackle, we should assign it to v13.0.0.

**Comment by jrpear** [comment]
I'll probably have it done by the end of this weekend. I've just got to understand the testing system to assess whether the failing tests should be modified or just removed (if they're testing something that doesn't make sense inside a docker container). And write a failing test which the change will fix.

**Comment by jrpear** [comment]
Seems that this was introduced by #952 to fix #947. Someone asked about it here and it was closed because it went stale: #978.

I can't figure out why the guy who opened the issue #947 wanted to be able to send mail to the root user. He might have just been playing around -- again, I can't imagine why you'd want to send mail to system users of the container. Status reports should be emailed to postmaster, not the root user. From #952: "The extra email is the virus notification email which is sent to postmaster and previously was not delivered"

Users have been working around this for quite a while: [#978](https://github.com/docker-mailserver/docker-mailserver/issues/978#issuecomment-399667653) [#2545](https://github.com/docker-mailserver/docker-mailserver/issues/2545#issuecomment-1097590082)

Knowing where the change came from, I feel comfortable reverting the breaking part of #952.

This comment should become outdated: https://github.com/docker-mailserver/docker-mailserver/blob/1076aac37d3e30e921f3917b7eb35b461904d1c4/target/scripts/helpers/postfix.sh#L78

**Comment by georglauterbach** [comment]
I will close this for now as #3264 has also been closed.
